### DISCUSSION TITLE
Convert series of parameter value files to a user supplied realization csv file?

### DISCUSSION THREAD

**Discussion by sjepsen395** [pypest/pyemu#522] [Q&A] [answered]
Dear PyEMU group,

I have a series of 20 parameter value files that I generated using the PEST lhs2pest utility (the "20" is just for a test):

parfile1.par
parfile2.par
.
.
.
parfile19.par
parfile20.par

Is there a tool in PyEMU or PEST that can convert these parameter value files to a "realization" csv file that can be run in PESTpp-IES or PESTpp-SWP? I'm doing single runs with noptmax=0. This realization file would have m+1 columns, where m = number of model parameters. The leftmost column is the name of the realization. I would input this to PESTpp-IES or PESTpp-SWP to compute objective functions at a range of parameter values.

Thanks,
Steve

**Comment by briochh** [comment, marked as answer]
@sjepsen395, there is an ensemble method in pyemu that might be able to help. `ParameterEnsemble.from_parfiles()`. You should be able to pass your Pst object and list of par files to create a ParameterEnsemble, you can then use all the to_...() methods to dump out to a file to use in ies or swp. 

Let us know how you get on.

**Reply by sjepsen395** [reply]
Thanks @briochh! This is what I had in mind.

Steve
